base/src/com/google/idea/blaze/base/sync/libraries/LibraryModifier.java

```diff
@@ -49,12 +44,12 @@ public Library.ModifiableModel getModifiableModel() {
   /** Writes the library content to its {@link Library.ModifiableModel}. */
   public void updateModifiableModel(BlazeProjectData blazeProjectData) {
     removeAllContents();
-    for (File classFile : libraryFilesProvider.getClassFiles(blazeProjectData)) {
-      addRoot(classFile, OrderRootType.CLASSES);
+    for (String classFileUrl : libraryFilesProvider.getClassFilesUrls(blazeProjectData)) {
+      addRoot(classFileUrl, OrderRootType.CLASSES);
     }
 
-    for (File sourceFile : libraryFilesProvider.getSourceFiles(blazeProjectData)) {
-      addRoot(sourceFile, OrderRootType.SOURCES);
+    for (String sourceFileUrl : libraryFilesProvider.getSourceFilesUrls(blazeProjectData)) {
```

Comment:
Then `clion` issue will be back since source detection logic is not visible from `base`. So I need to find a way to avoid returning `ImmutableList<File>` since it cannot contain neither `String` nor `VirtualFile` that are required for entries with dir in `jar`.